lib/smooth/core/filesystem/SDCard.cpp

```diff
@@ -17,56 +17,47 @@ limitations under the License.
 
 #include "smooth/core/logging/log.h"
 #include "smooth/core/filesystem/SDCard.h"
-#include "smooth/core/filesystem/FSLock.h"
 
 using namespace smooth::core::logging;
 
 namespace smooth::core::filesystem
 {
-    bool SDCard::do_common_initialization(const MountPoint& mount_point,
-                                          int max_file_count,
-                                          bool format_on_mount_failure,
-                                          void* slot_config)
-    {
-        esp_vfs_fat_sdmmc_mount_config_t mount_config{};
-        mount_config.format_if_mount_failed = format_on_mount_failure;
-        mount_config.max_files = max_file_count;
-
-        //mount_config.allocation_unit_size = 16 * 1024;
-
-        auto mount_result = esp_vfs_fat_sdmmc_mount((*mount_point).c_str(), &host, slot_config, &mount_config, &card);
+    static constexpr const char* log_tag = "SDCard";
 
-        initialized = mount_result == ESP_OK;
+    bool SDCard::do_common_error_checking(esp_err_t result)
+    {
+        initialized = result == ESP_OK;
 
         if (initialized)
         {
-            Log::info("SPISDCard", "SD Card initialized");
+            Log::info(log_tag, "Storage Card initialized");
             sdmmc_card_print_info(stdout, card);
         }
         else
         {
-            if (mount_result == ESP_FAIL)
+            if (result == ESP_FAIL)
             {
-                Log::error("SPISDCard", "Failed to mount the file system.");
+                Log::error(log_tag, "Failed to mount the file system.");
             }
             else
             {
-                Log::error("SPISDCard", "Failed to initialize SD Card.");
+                Log::error(log_tag, "Failed to initialize storage card.");
             }
         }
 
-        FSLock::set_limit(max_file_count);
-
         return initialized;
     }
 
+    // Note: If the SDCard is the only spi-device using spi-host-bus-master then the class calling this
+    // function should also call Master::deinitialize(spi_host_device_t spi_host) to deinitialize the
+    // spi-bus
```

Comment:
Can we make this automatic using ref a ref-counter?

Reply:
Will add reference counters to Master one for hspi and one for vspi.